arquillian-chameleon-runner/runner/src/main/java/org/arquillian/container/chameleon/runner/ChameleonTargetConfiguration.java

```diff
@@ -0,0 +1,197 @@
+package org.arquillian.container.chameleon.runner;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import org.arquillian.container.chameleon.api.ChameleonTarget;
+import org.arquillian.container.chameleon.api.Mode;
+import org.arquillian.container.chameleon.api.Property;
+import org.w3c.dom.Attr;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import static org.arquillian.container.chameleon.runner.RunnerExpressionParser.parseExpressions;
+
+public class ChameleonTargetConfiguration {
+
+    private static final int CONTAINER = 0;
+    public static final int VERSION = 1;
+    public static final int MODE = 2;
+
+    private String container;
+    private String version;
+    private Mode mode;
+
+    private Map<String, String> customProperties = new HashMap<>();
+
+    protected ChameleonTargetConfiguration(String value) {
+
+        final String[] definition = value.split(":");
+
+        if (definition.length != 3) {
+            throw new IllegalArgumentException("Value of Chameleon expression must be of form container:version:mode");
+        }
+
+        this.container = definition[CONTAINER];
+        this.version = definition[VERSION];
+        this.mode = Mode.valueOf(definition[MODE].toUpperCase());
+
+    }
+
+    protected ChameleonTargetConfiguration(String container, String version, Mode mode,
+        Map<String, String> customProperties) {
+        this.container = container;
+        this.version = version;
+        this.mode = mode;
+        this.customProperties = customProperties;
+    }
+
+
+    public static ChameleonTargetConfiguration from(ChameleonTarget chameleonTarget) {
+
+        if ("".equals(chameleonTarget.value())) {
```

Comment:
maybe `.trim()` combined with `.isEmpty()`?

Reply:
Yeah but with this way I am also protecting against `chameleonTarget` being null.